hdfs/client.py

```diff
@@ -145,14 +146,14 @@ class Client(object):
 
   __registry__ = {}
 
-  def __init__(self, url, root=None, proxy=None, timeout=None, session=None):
+  def __init__(self, url, root=None, proxy=None, timeout=None,
+               session=None, params=None):
```

Comment:
If we expose `params` as argument, we end up with the same ownership problem (on `params` instead of `session`). It also makes it more complex to figure out what final parameters get sent (e.g. we need remember that these take precedence over `session.params`).

Since the only parameter that the base client needs is `doauth`, can we cache its value as `_proxy` and add it to the final `params` in the `api_handler` method above (replacing the params `update`)?